mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -163,25 +187,28 @@ class StringIndexerModel (
     }
     transformSchema(dataset.schema, logging = true)
 
+    val metadata = NominalAttribute.defaultAttr
+      .withName($(outputCol)).withValues(labels).toMetadata()
+    // If we are skipping invalid records, filter them out.
+    val (filteredDataset, keepInvalid) = getHandleInvalid match {
```

Comment:
I'm OK with returning a tuple; that's a common pattern.  Do you mean that it makes the code inside the match statement confusing?